templatefitter/fit_model/parameter_handler.py

```diff
@@ -308,25 +294,31 @@ def conversion_vector(self) -> np.ndarray:
         assert self._floating_conversion_vector is not None
         return self._floating_conversion_vector
 
-    @property
-    def conversion_matrix(self) -> np.ndarray:
-        assert self._floating_conversion_matrix is not None
-        return self._floating_conversion_matrix
-
     def get_combined_parameters(
         self,
         parameter_vector: np.ndarray,
+        ncall: Optional[int] = None,
     ) -> np.ndarray:
-        return parameter_vector @ self.conversion_matrix + self.conversion_vector
+
+        if ncall == self.combined_parameters_cache[0]:
+            return self.combined_parameters_cache[1]
+        else:
+            zero_array = np.zeros(len(self.floating_parameter_mask))
+            zero_array[self.floating_parameter_mask] = parameter_vector
+            result = zero_array + self.conversion_vector
+            if ncall is not None:
+                self.combined_parameters_cache = (ncall, result)
+            return result
 
     def get_combined_parameters_by_index(
         self,
         parameter_vector: np.ndarray,
-        indices: Union[int, List[int]],
+        indices: Union[int, List[int], slice],
```

Comment:
Not sure if it is misleading if this function also accepts slices (as the one below also allows slicing) but this gives again a 6-8% speedup in my tests.